src/MqttPublishing.cpp

```diff
@@ -110,6 +133,61 @@ void MqttPublishingClass::publishField(std::shared_ptr<InverterAbstract> inv, ui
     }
 
     MqttSettings.publish(topic, String(inv->Statistics()->getChannelFieldValue(channel, fieldId)));
+
+    // topic = "W/{}/pvinverter/{}/{}".format(portalId, deviceId, key) # UPDATE THIS
+    // print("{} = {}".format(topic, data.get(key) ) )
+    // client.publish(topic, json.dumps({ "value": data.get(key) }) )
+    if (channel == 0) {
+
+        float fieldvalue = float(inv->Statistics()->getChannelFieldValue(channel, fieldId));
+        fieldvalue = round(fieldvalue * 100)/100;
+        String fieldname = (inv->Statistics()->getChannelFieldName(channel, fieldId));
+        String portalid = MqttSettings.getVictronPortalId();
+        if ( portalid == NULL ) { portalid = "NOportalId"; }
+        String topic = "W/" + portalid + "/pvinverter";
+        String topic_Victron_sum;
+        String topic_Victron_phase;
+    
+        char serial[sizeof(uint64_t) * 8 + 1];
+        snprintf(serial, sizeof(serial), "%0x%08x",
+        ((uint32_t)((inv->serial() >> 32) & 0xFFFFFFFF)),
+        ((uint32_t)(inv->serial() & 0xFFFFFFFF)));
+        String invSerial = String(serial);
```

Comment:
Use the inverter integrated serialString method instead of this snprintf

Reply:
done.